### ISSUE TITLE
input language information not available in BibleNLPBitextMining task

### ISSUE THREAD

**Issue by cmullovisl** [embeddings-benchmark/mteb#3635]
### Describe the bug

I'm trying to implement a custom encoder model that requires knowing the input language for each call to `encode`. The language information seems to be intended to be passed to `encode` via the `hf_subset` parameter. For bitext mining tasks the `hf_subset` value is fed to the `encode` call in [the evaluator](https://github.com/embeddings-benchmark/mteb/blob/2.3.0/mteb/_evaluators/text/bitext_mining_evaluator.py#L57) as the subset column name for "parallel" type datasets and as the `self.hf_subset` for non-parallel type datasets.
For the BibleNLPBitextMining task, which is of non-parallel type, this results in `hf_subset` value of e.g. "eng_Latn-aai_Latn" for both columns, resulting in loss of information which of the two columns is the current input language.
The column names for this task are "sentence1" and "sentence2" as a result from [the BibleNLPBitextMining task dataset transform](https://github.com/embeddings-benchmark/mteb/blob/2.3.0/mteb/tasks/bitext_mining/multilingual/bible_nlp_bitext_mining.py#L917-L924).

### To reproduce

```python
import logging
from torch.utils.data import DataLoader
import mteb
from mteb.types import BatchedInput, PromptType, Array
from mteb.abstasks.task_metadata import TaskMetadata
from mteb.models import EncoderProtocol
import numpy as np

logger = logging.getLogger(__name__)

class Model(EncoderProtocol):

    def encode(
        self,
        inputs: DataLoader[BatchedInput],
        task_metadata: TaskMetadata,
        hf_split: str,
        hf_subset: str,
        prompt_type: PromptType | None = None,
        **kwargs,
    ) -> np.ndarray:
        """Encodes the given sentences using the encoder.

        Args:
            sentences: The sentences to encode.
            task_name: The name of the task.
            prompt_type: The prompt type to use.
            **kwargs: Additional arguments to pass to the encoder.

        Returns:
            The encoded sentences.
        """
        logger.info(f"{hf_split=}")
        logger.info(f"{hf_subset=}")
        pass

logging.basicConfig(
    level=logging.INFO,
    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s'
)

model = Model("testmodel", "v0")
tasks = mteb.get_tasks(tasks=["BibleNLPBitextMining"], languages=["arb"])
results = mteb.evaluate(model, tasks, overwrite_strategy="always")
```

### Are you interested to contribute a fix for this bug?

Yes

**Comment by Samoed** [comment]
> The language information seems to be intended to be passed to encode via the hf_subset parameter

No, we're passing only information of current subset of task. You can get information of languages of current subset from `TaskMetadata` like `task_metadata.hf_subsets_to_langscripts[hf_subset]`.

Currently, for tasks you can do like
```python
class Model:
    input_num: int = 0

    def encode(...):
        subset_langs = task_metadata.hf_subsets_to_langscripts[hf_subset]
        cur_lang = subset_langs[self.input_num]
        ...
        input_num = (input_num + 1) % 2
```
but we don't guarantee that columns will be in the same order as in `hf_subset` languages.

Also for parallel datasets can be passed any column name as subset name, so this also won't guarantee that you will get correct language name (but currently they will pass language name as subset).

**Comment by cmullovisl** [comment]
Thank you for your suggested solution.  
Unfortunately it seems the dataset column order in the evaluator doesn't match the language order in `task_metadata.hf_subsets_to_langscripts`, e.g. for Danish:
```python
tasks = mteb.get_tasks(tasks=["BibleNLPBitextMining"], languages=["dan"])
print(tasks[0].metadata.hf_subsets_to_langscripts["dan_Latn-eng_Latn"])
# prints: ['dan-Latn', 'eng-Latn']
tasks[0].load_data()
print(tasks[0].dataset)
# prints:
# {'eng_Latn-dan_Latn': DatasetDict({
#     train: Dataset({
#         features: ['sentence1', 'sentence2'],
#         num_rows: 256
#     })
# }), 'dan_Latn-eng_Latn': DatasetDict({
#     train: Dataset({
#         features: ['sentence2', 'sentence1'],
#         num_rows: 256
#     })
# })}
```
but in the `encode` call English will be the first to be evaluated.

For the BibleNLPBitextMining task (which uses English-centric parallel data) the English column seems to always be the the first one to be evaluated, but I actually don't understand why. It seems like the column ordering is based on `self.sentences.features.keys()` in [`evaluator.__call__()`](blob/2.3.0/mteb/_evaluators/text/bitext_mining_evaluator.py#L41), but as far as I know Python does not provide any guarantees for the ordering of dictionary keys and the behaviour changes with different Python versions.


I ended up relying on fasttext language ID by choosing the language with the higher likelihood from the two candidates. This seems to work reasonably well, so I think the issue can be closed.
